### PR TITLE
Fix possibility for modders to execute emotes for everyone

### PR DESCRIPTION
**Pull request by yannikfolgmann**
Fixed possibility for modders to execute emotes for everyone

### CODE REVIEW

**Comment by TayMcKenzieNZ** [comment]
Nice 🙂 but I feel like perhaps it should be something like 5 unit distance to be safe, because you want to at least be able to do shared emotes with someone close to you.

**Comment by iSentrie** [comment]
2 units is enough, do use similar method for over a year, it's completely fine and no need for more.

**Comment by AvaN0x** [comment]
It would need to be 3 to match the client side
https://github.com/TayMcKenzieNZ/rpemotes/blob/master/client/Syncing.lua#L16

File: server/Server.lua [hunk 1]
```diff
@@ -3,10 +3,38 @@
...
+    if target == -1 then
+      return
+   end
+   local tped = GetPlayerPed(target)
+   local pedcoord = GetEntityCoords(ped)
+   local targetcoord = GetEntityCoords(tped)
+
+   local distance = #(pedcoord - targetcoord)
+
+   if distance > 2 then
```
Comment (iSentrie) [inline server/Server.lua:17]: as AvaN0x mentioned, client side use distance of 3, would be smart to match it

File: server/Server.lua [hunk 2]
```diff
@@ -3,10 +3,38 @@
...
```
Comment (iSentrie) [inline server/Server.lua:35]: same here

**Review by iSentrie — CHANGES_REQUESTED** [review]
check AvaN0x comment for details

File: server/Server.lua (hunk 2 above)
Comment (iSentrie) [inline server/Server.lua:35]:
```suggestion
   if distance > 3 then
```

File: server/Server.lua (hunk 1 above)
Comment (iSentrie) [inline server/Server.lua:17]: as AvaN0x mentioned, client side use distance of 3, would be smart to match it
```suggestion
   if distance > 3 then
```

**Comment by TayMcKenzieNZ** [comment]
Thanks guys 🙂

**Pull request merged**